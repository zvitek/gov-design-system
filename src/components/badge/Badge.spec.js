import { shallowMount } from '@vue/test-utils'
import GovBadge from '@components/badge/Badge'

let wrapper

describe('GovBadge', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovBadge)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovBadge')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should be inversed', () => {
        wrapper.setProps({
            inversed: true
        })
        expect(wrapper.classes()).toContain('gov-badge--inversed')
    })

    it('should be warning', () => {
        wrapper.setProps({
            warning: true
        })
        expect(wrapper.classes()).toContain('gov-badge--warning')
    })

    it('should be error', () => {
        wrapper.setProps({
            error: true
        })
        expect(wrapper.classes()).toContain('gov-badge--error')
    })
})
