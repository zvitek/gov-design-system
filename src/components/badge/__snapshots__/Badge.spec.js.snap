// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovBadge render correctly 1`] = `<span class="gov-badge"></span>`;
